main.tf

```diff
@@ -34,12 +34,16 @@ resource "aws_sns_topic_subscription" "this" {
   endpoint               = var.subscribers[each.key].endpoint
   endpoint_auto_confirms = var.subscribers[each.key].endpoint_auto_confirms
   raw_message_delivery   = var.subscribers[each.key].raw_message_delivery
-  redrive_policy = var.sqs_dlq_enabled ? coalesce(var.redrive_policy, jsonencode({
-    deadLetterTargetArn = join("", aws_sqs_queue.dead_letter_queue.*.arn)
-    maxReceiveCount     = var.redrive_policy_max_receiver_count
+  redrive_policy = var.sqs_dlq_enabled ? coalesce(var.redrive_policy, jsonencode(
+    var.redrive_policy_max_receiver_count != null ? {
+      deadLetterTargetArn = join("", aws_sqs_queue.dead_letter_queue.*.arn)
+      maxReceiveCount     = var.redrive_policy_max_receiver_count
+      } : {
+      deadLetterTargetArn = join("", aws_sqs_queue.dead_letter_queue.*.arn)
```

Comment:
```suggestion
    deadLetterTargetArn = join("", aws_sqs_queue.dead_letter_queue.*.arn)
    maxReceiveCount     = some_condition ? var.redrive_policy_max_receiver_count : null
```

Reply:
using anything other than what I have here still raises this error: 

```
Error: error creating SNS Topic Subscription: InvalidParameter: Invalid parameter: Attributes Reason: RedrivePolicy: invalid JSON. Attribute maxReceiveCount is unknown.
```

Reply:
Looks like this redrive policy may need to be added directly to the sqs queue

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sqs_queue

Then the redrive policy for sns topic subscription may only support the deadLetterTargetArn

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sns_topic_subscription#redrive_policy

Reply:
I don't think we need to remove `redrive_policy_max_receiver_count`. I think we need to add the appropriate `redrive_policy` to the appropriate resource.

`aws_sqs_queue` has both `deadLetterTargetArn` and `maxReceiveCount`

`aws_sns_topic_subscription` has only `deadLetterTargetArn`

Reply:
> aws_sqs_queue has both deadLetterTargetArn and maxReceiveCount

right, but the only sqs queue we're defining in the root module is the dlq - are you saying you want to include a redrive policy on it? That seems a little redundant, no?